openedx/adg/lms/applications/views.py

```diff
@@ -263,3 +266,100 @@ def get_context_data(self, request):
                     'birth_year': extended_profile.birth_date.year,
                 })
         return context
+
+
+class CoverLetterView(RedirectToLoginOrRelevantPageMixin, View):
+    """
+    View enabling the user to select a Business Line and upload or write a cover letter.
+    """
+
+    template_name = 'adg/lms/applications/cover_letter.html'
+    login_url = '/register'
+
+    def is_precondition_satisfied(self):
+        """
+        Checks if a written application is already submitted or not.
+        Returns:
+            bool: True if written application is not completed, False otherwise.
+        """
+        user_application_hub, _ = ApplicationHub.objects.get_or_create(user=self.request.user)
+
+        return not user_application_hub.is_written_application_completed
```

Comment:
This is exactly the duplicate of above `is_precondition_satisfied` and I believe same would be the pre condition of education and experience page, let's move it at a common place and use everywhere. OR we can simply move it in the mixin.